### ISSUE TITLE
web interface vs API

### ISSUE THREAD

**Issue by bakman2** [OctoPrint/OctoPrint-MalyanConnectionFix#4]
I am using this plugin and for my printer found some odd behaviour, but this might not be related to this plugin, what I noticed:

- printer is disconnected/powered down/off
- Turn the printer on, in the web interface, the "auto" option (which is default) and "connect", does not work, need to refresh the device list first, then I can click connect and the connection is made.

However:
When I use node-red and use the API and post the serial device, speed and connect; Octoprint connects automatically.

I notice the same behaviour with the TasmotaMQTT plugin, where a system call is used to connect `self._printer.connect` (that is from octoprint i assume (?)).

Long story short: the API connects correctly, the web interface does not, what is the difference between the two and is this difference correct ?

**Comment by foosel** [comment]
There is no difference per se, the web interface literally uses the API.

However the web interface will not automatically refresh the port list, and if the printer is not available on initial load the port will not show in there and only auto will be available. If port auto detect does not work for you and you need to specify the port (which you seen to do via nodered), there's your difference.

**Comment by bakman2** [comment]
Thanks for the quick response!
That makes sense, (in a way). The web interface does remove the interface when it disconnects. 

When I turn it back on and press connect:
```
Changing monitoring state from "Connecting" to "Offline"
Connection closed, closing down monitor
Changing monitoring state from "Offline" to "Detecting serial port"
Serial port list: ['/dev/ttyACM0', '/dev/ttyACM1']
Trying /dev/ttyACM0
Changing monitoring state from "Detecting serial port" to "Error: Connection error, see Terminal tab"
Unexpected error while connecting to serial port: AUTO IspError: ''Timeout'' @ comm.py:_openSerial:2661 (hook malyan_connection_fix)
```
 it is trying to connect to the the correct device: `/dev/ttyACM0`

This part works via node-red/API but not from the web interface, if they both use the API, there must be something different somewhere. 

It is not a big deal as I will continue to use node-red to control the printer, but it's not consistent.

**Comment by foosel** [comment]
AUTO -> try all serial ports that are detected, if something goes wrong there severely things will stop
specific serial port ("/dev/ttyACM0") -> only try that one

The Malyan printers are weird, I still don't understand why this plugin even makes connection with them work more reliably than without, I just created it after getting reports by users that it works.

That's all I can give you here. 

If you don't mind I'll close this since I don't see how this is even specific to this plugin. If you have a general problem with using OctoPrint you'd like to discuss I suggest the community forum.

**Comment by bakman2** [comment]
No worries, thanks for your awesome work! Fantastic stuff!
